cypher-shell/src/test/java/org/neo4j/shell/MainTest.java

```diff
@@ -84,18 +84,39 @@ public void connectMaybeInteractivelyPromptsForBothIfNone() throws Exception {
         PrintStream ps = new PrintStream(baos);
 
         Main main = new Main(inputStream, ps);
-        main.connectMaybeInteractively(shell, connectionConfig, true);
+        main.connectMaybeInteractively(shell, connectionConfig, true, true);
 
         String out = new String(baos.toByteArray(), StandardCharsets.UTF_8);
 
-        assertEquals( out, String.format( "username: bob%npassword: ******%n" ));
+        assertEquals(String.format( "username: bob%npassword: ******%n" ), out);
         verify(connectionConfig).setUsername("bob");
         verify(connectionConfig).setPassword("secret");
         verify(shell, times(2)).connect(connectionConfig);
     }
 
     @Test
-    public void connectMaybeInteractivelyDoesNotPromptIfNotInteractive() throws Exception {
+    public void promptsSilentlyForUsernameAndPasswordIfNoneGivenIfOutputRedirected() throws Exception {
+        doThrow(authException).doNothing().when(shell).connect(connectionConfig);
+
+        String inputString = "bob\nsecret\n";
```

Comment:
This is probably problematic on windows.

Reply:
It seems green on the Windows Unit Tester (and that line has been there for a long long time).